src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -568,10 +568,14 @@ def progress_callback(current, total):
             client.web_apps.sync_function_triggers_slot(resource_group_name, name, slot)
         else:
             client.web_apps.sync_function_triggers(resource_group_name, name)
-    except CloudError as ce:
+    except CloudError as ex:
         # This SDK function throws an error if Status Code is 200
-        if ce.status_code != 200:
-            raise ce
+        if ex.status_code != 200:
+            raise ex
+    except DefaultErrorResponseException as ex:
+        if ex.response.status_code != 200:
+            raise ex 
```

Comment:
I think your changes will throw more exceptions. Originally, it throws CloudError . Now, it will throw CloudError or DefaultErrorResponseException .

Reply:
Yeah the sdk function `sync_function_triggers` is already throwing a `DefaultErrorResponseException` so that would happen without this change anyways.

What's happening is that the sdk function `sync_function_triggers` expects the rest api call to sync triggers to return a 204. If it doesn't return a 204, it throws a `CloudError` exception. Turns out the rest api call to sync triggers  returns a 200 on success, not a 204 (not sure if this is a docs issue or what but I've tested it and it seems to be the case). So the sdk function `sync_function_triggers` throws an exception because even though it's a success, it's not 204.

Previously, this was a CloudError and we would catch that exception, and it it's a 200 status code we would ignore the exception because it was actually a success. At some point, I'm not sure when or why, the sdk function `sync_function_triggers` started throwing `DefaultErrorResponseException`s instead of `CloudError` exceptions for 200 status codes. This would result in the user seeing an exception from running config-zip and thinking it failed. So this just covers this new case of the sdk function throwing a `DefaultErrorResponseException` on success.